### ISSUE TITLE
No intersections using LineString

### ISSUE THREAD

**Issue by jv-soares** [divvitco/geodart#25]
It seems that `LineString.intersections()` is not properly calculating an intersection.

I have two `LineString` objects and I added them into https://geojson.io/ for better visualization. They clearly have one intersection point but I'm receiving an empty list when calling `LineString.intersections()`

<img width="408" alt="image" src="https://github.com/divvitco/geodart/assets/92590629/05c3f3b4-dfb1-4e71-be86-9f64ace34fbd">

Here the .csv for importing the map view: [points.csv](https://github.com/divvitco/geodart/files/14761089/points.csv)

**Comment by ntatko** [comment]
Your `points.csv` appears to be empty, but I've made a test which appears to fail using more complex samples.

```dart
test("intersections - complex", () {
      LineString line1 = LineString.fromJson({
        "type": "Feature",
        "properties": {},
        "geometry": {
          "coordinates": [
            [6.296572596843504, 51.7325811130674],
            [6.331468651390992, 51.712025343003575],
            [6.339382748143407, 51.7102129542283],
            [6.362602756171839, 51.70573600531736],
            [6.376359910921593, 51.698489083139606],
            [6.3797989059639235, 51.697743051348],
            [6.384957425902371, 51.6941191969093],
            [6.388224666178672, 51.69166785643759],
            [6.416596657291507, 51.679727882292354],
            [6.435167699648048, 51.6707707876628]
          ],
          "type": "LineString"
        }
      });
      LineString line2 = LineString.fromJson({
        "type": "Feature",
        "properties": {},
        "geometry": {
          "coordinates": [
            [6.319292233742146, 51.68612595532974],
            [6.336336178541103, 51.67900529039221],
            [6.37603045269816, 51.69859284173461],
            [6.4087075284441255, 51.70541424912682],
            [6.426250901758067, 51.684199715280386],
            [6.416617349164341, 51.67972214627187],
            [6.4071551961115745, 51.664153797567536]
          ],
          "type": "LineString"
        }
      });

      expect(line1.intersections(line2).features.length, 2);
    });
```

I'm working on a fix. I'll let you know when I find something that works.

**Comment by ntatko** [comment]
Released in 0.3.3
